### ISSUE TITLE
[xray] Unable to create 0 CPU actor when there are no slots available.

### ISSUE THREAD

**Issue by robertnishihara** [ray-project/ray#2588]
Start Python with `RAY_USE_XRAY=1 ipython`

```python
import ray

ray.init(num_cpus=4)

@ray.remote(num_cpus=1)
class Foo:
    def method(self):
        pass

fs = [Foo.remote() for _ in range(4)]

ray.get([x.method.remote() for x in f])  # This succeeds

@ray.remote
class Bar:
    def method(self):
        pass

b = Bar.remote()

ray.get(b.method.remote())  # This hangs
```

Reported by @ericl

**Comment by robertnishihara** [comment]
In the above setting, one issue is that `b.method.remote()` requires 1 CPU and so it won't be able to run. However, the following should run (replacing the actor with a task that requires 0 CPUs), but currently it does not.

```
@ray.remote(num_cpus=0)
def g():
    return 1

ray.get(g.remote())  # This hangs.
```

**Comment by guoyuhong** [comment]
Yes, @robertnishihara . Current actor resource management is [a lifetime resource](https://github.com/ray-project/ray/blob/master/src/ray/raylet/node_manager.cc#L1026-L1030), which means once an actor is created, it will occupy the resource until this actor is [disconnected](DisconnectClient). However, this disconnect message will be sent once the instance of an `ActorHandle` is garbage collected ([This code snippet](https://github.com/ray-project/ray/blob/master/python/ray/actor.py#L863-L875) shows that only in `__del__` of `ActorHandle` the `DisconnectClient` message will be send.). And there is no warning when this case happens.

In my understanding, precious resource should not be determined by garbage collection behavior. I think we can provide a `release` function like the close function in file handle of python.
